packages/amplify-graphql-schema-generator/src/datasource-adapter/mysql-datasource-adapter.ts

```diff
@@ -80,7 +91,14 @@ export class MySQLDataSourceAdapter extends DataSourceAdapter {
   }
 
   public async getTablesList(): Promise<string[]> {
-    const result = (await this.dbBuilder.raw("SHOW TABLES"))[0];
+    const SHOW_TABLES_QUERY = `SHOW TABLES`;
+    let result;
+    if (this.useVPC) {
+      result = await invokeSchemaInspectorLambda(this.vpcSchemaInspectorLambda!, this.config, SHOW_TABLES_QUERY, this.vpcLambdaRegion);
```

Comment:
just a thought here, can we refactor this to
```
if (this.vpcSchemaInspectorLambda) {
...
}
```
and get rid of the `!` and a boolean flag?
I see the same pattern in other places in this file as well.

Reply:
Added `vpcSchemaInspectorLambda` to the if condition so that we can get rid of ! check. I would like to keep the boolean flag to explicitly required to use the VPC lambda.